src/openforms/authentication/contrib/generic_oidc/views.py

```diff
@@ -0,0 +1,139 @@
+from typing import TypedDict, override
+
+from django.http import HttpRequest
+from django.http.response import HttpResponseRedirect
+
+from digid_eherkenning.oidc.models import BaseConfig
+from digid_eherkenning.oidc.views import (
+    OIDCAuthenticationCallbackView as _OIDCAuthenticationCallbackView,
+)
+from furl import furl
+from mozilla_django_oidc_db.config import lookup_config
+from mozilla_django_oidc_db.views import _RETURN_URL_SESSION_KEY, OIDCInit
+
+from openforms.authentication.contrib.digid.views import (
+    DIGID_MESSAGE_PARAMETER,
+    LOGIN_CANCELLED,
+)
+from openforms.authentication.contrib.eherkenning.views import (
+    MESSAGE_PARAMETER as EH_MESSAGE_PARAMETER,
+)
+from openforms.authentication.contrib.yivi_oidc.constants import YIVI_MESSAGE_PARAMETER
+
+from ...views import BACKEND_OUTAGE_RESPONSE_PARAMETER
+
+
+class OptionsT(TypedDict):
+    pass
+
+
+class OIDCAuthenticationInitView[OptionsT](OIDCInit):
+    options: OptionsT
+
+    @override
+    def get(
+        self,
+        request: HttpRequest,
+        return_url: str = "",
+        options: OptionsT = None,
+        *args,
+        **kwargs,
+    ) -> HttpResponseRedirect:
+        if options is None:
+            options = {}
+        self.options = options
+        return super().get(request, return_url, *args, **kwargs)
```

Comment:
See my comment at https://github.com/open-formulieren/open-forms/pull/5363/files#diff-722534ade5c574db4a55aa7b240672888840d3d20ab27425816fd886df38d01fR71 for how you can avoid having to do this override entirely. Check the base class `OIDCInit` implementation to figure out what's needed to make this work via `.as_view(...)` and the django docs of course: https://docs.djangoproject.com/en/5.2/topics/class-based-views/#usage-in-your-urlconf